src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/config/GerritJcascConfigurator.java

```diff
@@ -0,0 +1,309 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.sonyericsson.hudson.plugins.gerrit.trigger.config;
+
+import com.google.common.collect.ImmutableSet;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.GerritServer;
+import com.sonyericsson.hudson.plugins.gerrit.trigger.PluginImpl;
+import com.sonymobile.tools.gerrit.gerritevents.watchdog.WatchTimeExceptionData;
+import hudson.Extension;
+import io.jenkins.plugins.casc.Attribute;
+import io.jenkins.plugins.casc.BaseConfigurator;
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.Configurator;
+import io.jenkins.plugins.casc.ConfiguratorException;
+import io.jenkins.plugins.casc.ConfiguratorRegistry;
+import io.jenkins.plugins.casc.impl.attributes.MultivaluedAttribute;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import io.jenkins.plugins.casc.model.Sequence;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Configure JCasC.
+ */
+@Restricted(NoExternalUse.class)
+@Extension
+public class GerritJcascConfigurator extends BaseConfigurator<PluginImpl> {
+
+    /**
+     * Empty constructor.
+     */
+    public GerritJcascConfigurator() {
+        super();
+    }
+
+    @Override
+    protected PluginImpl instance(Mapping mapping, ConfigurationContext configurationContext) {
+        return PluginImpl.getInstance();
+    }
+
+    @Override
+    @Nonnull
+    public String getName() {
+        return PluginImpl.SYMBOL_NAME;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return PluginImpl.DISPLAY_NAME;
+    }
+
+    @Override
+    public Class<PluginImpl> getTarget() {
+        return PluginImpl.class;
+    }
+
+    @Override
+    protected void configure(
+        Mapping config, PluginImpl instance, boolean dryrun, ConfigurationContext context
+    ) throws ConfiguratorException {
+        List<GerritServer> oldServers = instance.getServers();
+
+        super.configure(config, instance, dryrun, context);
+
+        instance.getPluginConfig().updateEventFilter();
+
+        for (GerritServer oldServer : oldServers) {
+            oldServer.stopConnection();
+            oldServer.stop();
+        }
+
+        // Mimicking GerritManagement#doAddNewServer
+
+        List<String> serverNames = new ArrayList<>();
+        for (GerritServer server : instance.getServers()) {
+            String name = server.getName();
+            if (serverNames.contains(name)) {
+                throw new ConfiguratorException(this, "Multiple gerrit servers with name: " + name);
```

Comment:
This will sort of leave the configuration in a bad state. The verification should be done before setting the new values, somehow...